utils/jwt_with_key_rotation.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import * as jwt from "jsonwebtoken";
+import { ulid } from "ulid";
+
+import { pipe } from "fp-ts/lib/function";
+import * as TE from "fp-ts/TaskEither";
+import * as E from "fp-ts/Either";
+import * as O from "fp-ts/Option";
+
+import { Second } from "@pagopa/ts-commons/lib/units";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+
+// --------------------
+
+const alg = "RS256";
+
+// -----------------------
+// Public methods
+// -----------------------
+
+/**
+ * JWT Generation
+ */
+export type GetGenerateJWT = <T extends Record<string, unknown>>(
+  issuer: NonEmptyString,
+  primaryPrivateKey: NonEmptyString
+) => (payload: T, ttl: Second) => TE.TaskEither<Error, NonEmptyString>;
+
+export const getGenerateJWT: GetGenerateJWT = (
+  issuer,
+  primaryPrivateKey
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+) => (payload, ttl) =>
```

Comment:
Can we explicit return type here?

Reply:
Done in [[#IOCIT-301] remove DecodedAuthJWT + minor fix](https://github.com/pagopa/io-functions-lollipop/pull/14/commits/915afe8d302a5022519a7819395f187c71a9c485)